src/runtime/reference.ts

```diff
@@ -93,11 +98,21 @@ export class Reference implements Storable {
     };
   }
 
+  static abstractOutBackingKey(storageKey: string): string {
```

Comment:
This name is a bit confusing. It's not clear what the "abstract" bit is referring to. Should it be `extractBackingKey`?